qemu/tests/rh_qemu_update.py

```diff
@@ -0,0 +1,261 @@
+        """
+        def nice_format(ugly_format):
+            better_format = ugly_format.replace("  ", " ")
+            if "  " in better_format:
+                return nice_format(better_format)
+            else:
+                return better_format
+        build_or_module = "cm_cmd_brew"
+        if install_id.isnumeric():
+            build_or_module = "cm_cmd_module"
+        # Prepare module or build repo containing newer version of qemu-kvm
+        cmd = f"python3 {params['cm_path']}{params[build_or_module]} {install_id}"
+        test.log.info(f"Running: {nice_format(cmd)}")
+        try:
+            session.cmd(cmd, timeout=OPERATION_TIMEOUT)
+            test.log.info("Creation of repo '{}' succeeded".format(
+                install_id
+                )
+            )
+        except Exception as e:
+            test.error("Creation of repo '{}' failed with: {}".format(
+                install_id, e
+                )
+            )
+        # Disable and enable new module if module is used
+        if install_id.isnumeric():
+            stream = _get_module_stream(session, install_id)
+            disable_cmd = f"{params['pm_tool']} module --disable virt:{stream}"
+            s_disable, o_disable = session.cmd_status_output(
+                disable_cmd, timeout=OPERATION_TIMEOUT
+                )
+            if s_disable != 0:
+                test.fail("Disable of module virt failed with: {}".format(
+                    o_disable
+                    )
+                )
+            else:
+                test.log.info("Disable of module virt succeeded")
+            enable_cmd = f"{params['pm_tool']} module --enable virt:{stream}"
+            s_enable, o_enable = session.cmd_status_output(
+                enable_cmd, timeout=OPERATION_TIMEOUT
+                )
+            if s_enable != 0:
+                test.fail("Enable of module virt:{} failed with: {}".format(
+                    stream, o_enable
+                    )
+                )
+            else:
+                test.log.info("Enable of module virt:{} succeeded".format(
+                    stream
+                    )
+                )
+        # Run upgrade to newer qemu-kvm version
+        cmd_upgrade = "{} upgrade -y {}".format(
+            params['pm_tool'], qemu_package
+        )
+        s_upgrade, o_upgrade = session.cmd_status_output(
+            cmd_upgrade, timeout=INSTALL_TIMEOUT)
+        if s_upgrade != 0:
+            test.error("Upgrade of '{}' failed with: {}".format(
+                qemu_package, o_upgrade
+                )
+            )
+        test.log.info("Upgrade of '{}' succeeded".format(
+            qemu_package
+            )
+        )
+
+    def verify_installed_qemu(session):
+        """
+        Verify installated version of qemu-kvm matches expected one by its NVR
+        """
+        expected_nvr = _get_installed_qemu_info()["nvr"]
```

Comment:
Duplicate work with `verify_qemu_version`. how do you think about:
```
host_qemu =  get_installed_qemu_info()
guest_qemu_before = get_installed_qemu_info(session)
update_guest_qemu()  ==> include checking if host > guest
guest_qemu_after = get_installed_qemu_info(session)
verify_installed_qemu()
```

Reply:
I see your point. I slightly modified it into:
```
    host_qemu = _get_installed_qemu_info()
    guest_qemu_before = _get_installed_qemu_info(session)
    install_id = verify_qemu_version(host_qemu, guest_qemu_before)
    update_guest_qemu(session, install_id)
    guest_qemu_after = _get_installed_qemu_info(session)
    verify_installed_qemu(host_qemu, guest_qemu_after)
```